### PR TITLE
core/tracker: add tracker component - WIP

### PR DESCRIPTION
**Pull request by xenowits**
Adds tracker component

category: feature
ticket: #768

### CODE REVIEW

File: core/tracker/tracker.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	"github.com/obolnetwork/charon/app/log"
+	"github.com/obolnetwork/charon/app/z"
+	"github.com/obolnetwork/charon/core"
+)
+
+// Component refers to a core workflow component.
+type Component int
+
+// Core components arranged in the order data flows through them.
+const (
```
Comment (xenowits) [inline core/tracker/tracker.go:32]: I'm not perfectly sure if this is the correct order for the components.

File: core/tracker/tracker.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+		deadlineFunc: deadlineFunc,
+	}
+
+	return t
+}
+
+// Run blocks and registers events from each component in input channel. It also checks for duty deadline every 2s.
+// If deadline exceeds for a duty, it instruments the duty and removes it from the events map.
+func (t *Tracker) Run(ctx context.Context) error {
+	timer := time.NewTicker(1 * time.Second)
```
Comment (xenowits) [inline core/tracker/tracker.go:88]: `timer` will be replaced with something like a `slotTicker`

File: core/tracker/tracker.go [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+					}
+				}
+			}
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+}
+
+// DutyDB, ValidatorAPI, ParSigDB, AggSigDB and Broadcast components don't provide subscriptions.
```
Comment (xenowits) [inline core/tracker/tracker.go:121]: Need to find a way to register subscriptions for these components

File: core/tracker/tracker_internal_test.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	"github.com/stretchr/testify/require"
+
+	"github.com/obolnetwork/charon/app/log"
+	"github.com/obolnetwork/charon/core"
+	"github.com/obolnetwork/charon/core/fetcher"
+	"github.com/obolnetwork/charon/testutil"
+	"github.com/obolnetwork/charon/testutil/beaconmock"
+)
+
+func TestTracker_FetcherEvent(t *testing.T) {
```
Comment (xenowits) [inline core/tracker/tracker_internal_test.go:33]: This just tests if tracker gets events from a single component `Fetcher`. Need to write test to check if tracker receives events from all the components

**Comment by dB2510** [comment]
still needs some work

**Pull request closed without merging**